client/data/help/use-help-search-query.ts

```diff
@@ -1,26 +1,90 @@
-import { useLocale } from '@automattic/i18n-utils';
-import { useQuery } from '@tanstack/react-query';
-import wpcomRequest from 'wpcom-proxy-request';
-import type { LinksForSection } from '@automattic/data-stores';
+import { useQueryClient, useQuery } from '@tanstack/react-query';
+import apiFetch from '@wordpress/api-fetch';
+import wpcomRequest, { canAccessWpcomApis } from 'wpcom-proxy-request';
+
+export interface SearchResult {
+	link: string;
+	title: string | React.ReactChild;
+	content?: string;
+	icon?: string;
+	post_id?: number;
+	blog_id?: number;
+	source?: string;
+}
+
+interface APIFetchOptions {
+	global: boolean;
+	path: string;
+}
+
 export const useHelpSearchQuery = (
 	search: string,
-	queryOptions: Record< string, unknown > = {}
+	locale = 'en',
+	queryOptions: Record< string, unknown > = {},
+	sectionName = ''
 ) => {
-	const locale = useLocale();
-	const params = new URLSearchParams( {
-		include_post_id: '1',
-		locale,
-	} );
+	const queryClient = useQueryClient();
 
-	if ( search ) {
-		params.append( 'query', search );
+	// set up article list fetching
+	let fetchArticlesAPI: () => Promise< unknown >;
+	const params = `?query=${ encodeURIComponent( search ) }&locale=${ encodeURIComponent(
+		locale
+	) }&section=${ encodeURIComponent( sectionName ) }`;
+	if ( canAccessWpcomApis() ) {
+		fetchArticlesAPI = () =>
```

Comment:
I would make `fetchArticlesAPI` a function that has these `ifs` and `elses` inside it. In which case, you can put it outside the hook to avoid recreating it with every render.

```js
function fetchArticlesAPI() {
	if ( canAccessWpcomApis() ) {
		return wpcomRequest( {
				path: `help/search/wpcom${ params }`,
				apiNamespace: 'wpcom/v2/',
				apiVersion: '2',
		} );	
	} else {		
		return apiFetch( {
				global: true,
				path: `/help-center/search${ params }`,
			} as APIFetchOptions );
	}
}
```

Reply:
Thanks!  implemented